test/webaudio/audioworklet_emscripten_locks.cpp

```diff
@@ -1,26 +1,27 @@
+#include <emscripten/wasm_worker.h>
 #include <emscripten/webaudio.h>
-#include <emscripten/threading.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
 
-// Tests that
-// - _emscripten_thread_supports_atomics_wait() returns true in a Wasm Audio Worklet.
-// - emscripten_futex_wake() does not crash in a Wasm Audio Worklet.
-// - emscripten_futex_wait() does not crash in a Wasm Audio Worklet.
+// Tests that these audio worklet compatible functions work:
+// - emscripten_lock_init()
+// - emscripten_lock_busyspin_wait_acquire()
+// - emscripten_lock_busyspin_waitinf_acquire()
+// - emscripten_lock_try_acquire()
+// - emscripten_lock_release()
 // - emscripten_get_now() does not crash in a Wasm Audio Worklet.
 
-int futexLocation = 0;
+// Todo: add/find a way to verify atomic wait fails?
+
+emscripten_lock_t testLock = EMSCRIPTEN_LOCK_T_STATIC_INITIALIZER;
 int testSuccess = 0;
 
 bool ProcessAudio(int numInputs, const AudioSampleFrame *inputs, int numOutputs, AudioSampleFrame *outputs, int numParams, const AudioParamFrame *params, void *userData) {
-  int supportsAtomicWait = _emscripten_thread_supports_atomics_wait();
-  printf("supportsAtomicWait: %d\n", supportsAtomicWait);
-  assert(!supportsAtomicWait);
```

Comment:
Why not keep these three lines around?

Reply:
I don’t see `_emscripten_thread_supports_atomics_wait()` in a public API.

Non-support of wait is a given for AW global contexts, which is why I didn’t think it worth the effort of exposing the function.

Reply:
I thought about this, and I’ll expose the function publicly. I can think of a few use cases, sharing code between threads that can or cannot wait, for example.

Reply:
Made `emscripten_thread_supports_atomics_wait()` public (added with the other Emscripten threading functions).